src/main/java/com/celements/web/service/CelementsWebScriptService.java

```diff
@@ -1100,4 +1103,28 @@ public String getSupportLinkURL() {
     return url;
   }
 
+  public String encodeUrlToUtf8(String urlStr) {
+    String pattern = "://";
+    int findIndex = urlStr.indexOf(pattern);
+    if (findIndex > 0) {
+      String urlPrefix = urlStr.substring(0, findIndex + pattern.length());
+      String mainUrl = urlStr.substring(findIndex + pattern.length());
+      try {
+        urlStr = URLEncoder.encode(mainUrl, "UTF-8");
+      } catch (UnsupportedEncodingException exp) {
+        LOGGER.error("Failed to encode url [" + urlStr + "] to utf-8", exp);
+      }
+      urlStr = urlPrefix + urlStr;
+    }
+    return urlStr;
+  }
+
+  public void sendRedirect(String urlStr) {
```

Comment:
please move implementation to xxxService and only add a delegation in the ScriptService